generators/php/php_generator.ts

```diff
@@ -242,65 +251,62 @@ export class PhpGenerator extends CodeGenerator {
-   * @param {!Block} block The block.
-   * @param {string} atId The property ID of the element to get.
-   * @param {number=} opt_delta Value to add.
-   * @param {boolean=} opt_negate Whether to negate the value.
-   * @param {number=} opt_order The highest order acting on this value.
-   * @return {string|number}
+   * Generate code representing the specified value input, adjusted to take into
+   * account indexing (zero- or one-based) and optionally by a specified delta
+   * and/or by negation.
+   *
+   * @param block The block.
+   * @param atId The ID of the input block to get (and adjust) the value of.
+   * @param delta Value to add.
+   * @param negate Whether to negate the value.
+   * @param order The highest order acting on this value.
+   * @returns The adjusted value.
    */
-  getAdjusted(block, atId, opt_delta, opt_negate, opt_order) {
-    let delta = opt_delta || 0;
-    let order = opt_order || this.ORDER_NONE;
+  getAdjusted(block: Block, atId: string, delta = 0, negate = false, order = Order.NONE): string | number {
     if (block.workspace.options.oneBasedIndex) {
       delta--;
     }
     let defaultAtIndex = block.workspace.options.oneBasedIndex ? '1' : '0';
-    let outerOrder = order;
-    let innerOrder;
+
+    let orderForInput = order;
     if (delta > 0) {
-      outerOrder = this.ORDER_ADDITION;
-      innerOrder = this.ORDER_ADDITION;
+      orderForInput = Order.ADDITION;
     } else if (delta < 0) {
-      outerOrder = this.ORDER_SUBTRACTION;
-      innerOrder = this.ORDER_SUBTRACTION;
-    } else if (opt_negate) {
-      outerOrder = this.ORDER_UNARY_NEGATION;
-      innerOrder = this.ORDER_UNARY_NEGATION;
+      orderForInput = Order.SUBTRACTION;
+    } else if (negate) {
+      orderForInput = Order.UNARY_NEGATION;
     }
-    let at = this.valueToCode(block, atId, outerOrder) || defaultAtIndex;
 
-    if (stringUtils.isNumber(at)) {
-      // If the index is a naked number, adjust it right now.
-      at = Number(at) + delta;
-      if (opt_negate) {
-        at = -at;
-      }
-    } else {
-      // If the index is dynamic, adjust it in code.
-      if (delta > 0) {
-        at = at + ' + ' + delta;
-      } else if (delta < 0) {
-        at = at + ' - ' + -delta;
-      }
-      if (opt_negate) {
-        if (delta) {
-          at = '-(' + at + ')';
-        } else {
-          at = '-' + at;
-        }
-      }
-      innerOrder = Math.floor(innerOrder);
-      order = Math.floor(order);
-      if (innerOrder && order >= innerOrder) {
-        at = '(' + at + ')';
+    let at = this.valueToCode(block, atId, orderForInput) || defaultAtIndex;
+
+    // Easy case: no adjustments.
+    if (delta === 0 && !negate) {
+      return at;
+    }
+    // If the index is a naked number, adjust it right now.
+     if (stringUtils.isNumber(at)) {
+      at = String(Number(at) + delta);
+      if (negate) {
+        at = String(-Number(at));
```

Comment:
I guess `at` has to be a string for TS since `valueToCode` will return a string. but the converting back and forth between string and number is really ugly. would it look better if you used a different variable and only convert it to string when you're returning that value?

Reply:
A good suggestion.  I think this is best dealt with separately, however, so that all the implementations can be improved at the same time and perhaps made more consistent in the process, so I've again added it to the list in #7600.